src/coreComponents/mesh/mpiCommunications/PartitionerBase.hpp

```diff
@@ -0,0 +1,193 @@
+   * @return a static reference to the Catalog object
+   */
+  static CatalogInterface::CatalogType & getCatalog();
+
+  /**
+   * @brief Partition a mesh according to its dual graph.
+   * @param graph the input graph (edges of locally owned nodes)
+   * @param vertDist the parallel distribution of vertices: vertex index offset on each rank
+   * @param numParts target number of partitions
+   * @param comm the MPI communicator of processes to partition over
+   * @param numRefinements number of partition refinement iterations
+   * @return an array of target partitions for each element in local mesh
+   */
+  virtual array1d< pmet_idx_t > partition( ArrayOfArraysView< pmet_idx_t const, pmet_idx_t > const & graph,
+                                           arrayView1d< pmet_idx_t const > const & vertDist,
+                                           pmet_idx_t const numParts,
+                                           MPI_Comm comm,
+                                           int const numRefinements ) = 0;
+
+  virtual void setPartitionCounts( unsigned int xPartitions,
+                                   unsigned int yPartitions,
+                                   unsigned int zPartitions ) = 0;
+
+
+  virtual void setNeighborsRank( const std::vector< int > & neighborsRank ) = 0;
+
+  std::vector< int > const & getNeighborsRank() const
+  {
+    return m_neighborsRank;
+  }
+
+  std::vector< int > & getNeighborsRank()
+  {
+    return m_neighborsRank;
+  }
+
+
+  void buildNeighbors();
+
+
+  stdVector< NeighborCommunicator > & getNeighbors()
+  { return m_neighbors; }
+
+  stdVector< NeighborCommunicator > const & getNeighbors() const
+  { return m_neighbors; }
+
+
+  int getNumNeighbors()
+  {
+    return static_cast< int >(m_neighbors.size());
+  }
+
+  int getNumNeighbors() const
+  {
+    return static_cast< int >(m_neighbors.size());
+  }
+
+
+  int getNumFirstOrderNeighbors()
+  {
+    return m_numFirstOrderNeighbors;
+  }
+
+  int getNumFirstOrderNeighbors() const
+  {
+    return m_numFirstOrderNeighbors;
+  }
+
+
+
+  stdVector< NeighborCommunicator > getFirstOrderNeighbors()
+  {
+    return stdVector< NeighborCommunicator >(
+      m_neighbors.begin(),
+      m_neighbors.begin() + std::min( m_numFirstOrderNeighbors, getNumNeighbors())
+      );
+  }
+
+  stdVector< NeighborCommunicator > getFirstOrderNeighbors() const
+  {
+    return stdVector< NeighborCommunicator >(
+      m_neighbors.begin(),
+      m_neighbors.begin() + std::min( m_numFirstOrderNeighbors, getNumNeighbors())
+      );
+  }
+
+
+  void appendNeighborsOfNeighbors();
+
+
+  virtual void color() = 0;
+
+  int getColor() const
+  {
+    return m_color;
+  }
+  int getColor()
+  {
+    return m_color;
+  }
+
+  int getNumColors() const
+  {
+    return m_numColors;
+  }
+  int getNumColors()
+  {
+    return m_numColors;
+  }
+
+
+  int getNumPartitions() const
+  {
+    return m_numPartitions;
+  }
+  int getNumPartitions()
+  {
+    return m_numPartitions;
+  }
```

Comment:
you don't need `non-const getters`. It would make sense if you were returning a reference but I would not do that and add a setter instead if you need one.